sql/core.go

```diff
@@ -345,6 +346,17 @@ type CheckAlterableTable interface {
 	DropCheck(ctx *Context, chName string) error
 }
 
+// PrimaryKeyAlterableTable represents a table that supports primary key changes.
+type PrimaryKeyAlterableTable interface {
+	Table
+	// CreatePrimaryKey creates a primary key for this table, using the provided parameters.
+	// Returns an error if the new primary key set is not compatible with the current table data
+	// or the table already has a key.
+	CreatePrimaryKey(ctx *Context, columns []string) error
```

Comment:
Copy the interface for `IndexAlterableTable` here, but with no indexName param:

`CreateIndex(ctx *Context, indexName string, using IndexUsing, constraint IndexConstraint, columns []IndexColumn, comment string) error`

Reply:
Does mysql even enforce names primary key constraints? When I was testing the given constraint name was never used in the information_schema.table_constraints name. Seems like this is just adding additional params that will never need to be used.

Reply:
After discussing it seems like IndexColumn is not enough. indexName, using, and index constraint are not needed